throttle/contract_tx_limiter_middleware.go

```diff
@@ -0,0 +1,124 @@
+package throttle
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/loomnetwork/go-loom"
+	"github.com/loomnetwork/go-loom/plugin/contractpb"
+	"github.com/loomnetwork/loomchain"
+	"github.com/loomnetwork/loomchain/auth"
+	udw "github.com/loomnetwork/loomchain/builtin/plugins/user_deployer_whitelist"
+	"github.com/loomnetwork/loomchain/vm"
+	"github.com/pkg/errors"
+)
+
+const (
+	deployerStatePrefix = "ds"
+)
+
+type contractTxLimiter struct {
+	// contract_address to limiting parametres structure
+	contractToTierMap map[string]udw.Tier
+	// track of no. of txns in previous blocks per contract ()
+	contractToBlockTrx map[string]map[int64]int64
+}
+
+var TxLimiter *contractTxLimiter
+
+func (txl *contractTxLimiter) isAccountLimitReached(contractAddr loom.Address, curBlockHeight int64) bool {
+	tier, ok := txl.contractToTierMap[contractAddr.String()]
+	if !ok {
+		return false
+	}
+	blockTxns, ok := txl.contractToBlockTrx[contractAddr.String()]
+	if !ok {
+		return false
+	}
+	minBlockHeight := curBlockHeight - int64(tier.BlockRange) + 1
```

Comment:
handle edge cases here, like `curBlockHeight` is less than `BlockRange`

Reply:
Also, I feel this algo can be written in more concise format. will get back to you with exact idea.

Reply:
even if curBlockHeight is less than BlockRange, still there wont be any issue ... it is int64 ..